cpp/src/io/orc/reader_impl.cu

```diff
@@ -178,15 +178,28 @@ class metadata {
    * @brief Filters and reads the info of only a selection of stripes
    *
    * @param[in] stripe Index of the stripe to select
+   * @param[in] max_stripe_count Number of stripes to select for stripe-based selection
+   * @param[in] stripe_indices Indices of individual stripes [max_stripe_count]
    * @param[in] row_start Starting row of the selection
    * @param[in,out] row_count Total number of rows selected
    *
    * @return List of stripe info and total number of selected rows
    **/
-  auto select_stripes(int stripe, int max_stripe_count, int &row_start, int &row_count) {
+  auto select_stripes(int stripe, int max_stripe_count, const size_type *stripe_indices,
```

Comment:
I think we generally to to avoid `int` a prefer the `int32_t` (multiple places in this PR)
```suggestion
  auto select_stripes(int32_t stripe, int32_t max_stripe_count, const size_type *stripe_indices,
```

Reply:
These were pre-existing (should be size_type, I suppose)

Reply:
If it refers to a size of a cudf::column, then `size_type`.

Reply:
It's mainly rowgroup units, but they are indeed declared as size_type in the API so should match there (I believe that particular code was written before size_type existed) ~ will make the change.